GVFS/GVFS.Common/LocalGVFSConfig.cs

```diff
@@ -0,0 +1,99 @@
+﻿using GVFS.Common.FileSystem;
+using GVFS.Common.Git;
+using System;
+using System.IO;
+
+namespace GVFS.Common
+{
+    public class LocalGVFSConfig
+    {
+        private const string FileName = "local_config.dat";
+        private string configFile;
+        private string gitPath;
+        private PhysicalFileSystem fileSystem;
+
+        public LocalGVFSConfig(string gitPath)
+        {
+            string servicePath = Paths.GetServiceDataRoot(GVFSConstants.Service.ServiceName);
+            string gvfsDirectory = Path.GetDirectoryName(servicePath);
+
+            this.configFile = Path.Combine(gvfsDirectory, FileName);
+            this.gitPath = gitPath;
+            this.fileSystem = new PhysicalFileSystem();
+        }
+
+        public bool TryGetValueForKey(string key, out string value, out string error)
+        {
+            return this.TryGetConfig(this.KeyWithGVFSSectionPrefix(key), out value, out error);
+        }
+
+        public bool TrySetValueForKey(string key, string value, out string error)
+        {
+            return this.TrySetConfig(this.KeyWithGVFSSectionPrefix(key), value, out error);
+        }
+
+        private bool TryGetConfig(string key, out string value, out string error)
+        {
+            if (!this.fileSystem.FileExists(this.configFile))
+            {
+                error = $"Error reading {key}. Config file({this.configFile}) does not exist.";
+                value = null;
+                return false;
+            }
+
+            GitProcess.Result result = GitProcess.GetFromFileConfig(this.gitPath, this.configFile, key);

```

Comment:
This is a definite downside to using git's format - we have to launch a process every time we want to read or write a value. Using our own file format would not have that issue.

Reply:
Ya... replaced with FileBasedDictionary